src/System.ObjectModel/src/System/Collections/ObjectModel/ObservableCollection.cs

```diff
@@ -123,9 +126,61 @@ protected override void RemoveItem(int index)
 
             OnCountPropertyChanged();
             OnIndexerPropertyChanged();
-            OnCollectionChanged(NotifyCollectionChangedAction.Remove, removedItem, index);
+
+            if (!_skipCollectionChanged)
+            {
+                OnCollectionChanged(NotifyCollectionChangedAction.Remove, removedItem, index);
+            }
         }
 
+        /// <summary>
+        /// Called by base class Collection&lt;T&gt; when a count of items is removed from the list;
+        /// raises a CollectionChanged event to any listeners.
+        /// </summary>
+        protected override void RemoveItemsRange(int index, int count)
```

Comment:
Do we expect events to be raised if `count` is `0`?

Reply:
After looking at the different scenarios if `count` is `0` it will make it through the `Collection<T>` validation checks and back to this method. I agree we should ignore the events if count = 0 because nothing really happened.

Options:

1. Add check at top of method:
```c#
protected override void RemoveItemsRange(int index, int count)
{
    if (count == 0) return;
    // omitted code
}
```

2. Add check to ignore raising events
```c#
protected override void RemoveItemsRange(int index, int count)
{
    // omitted code
  
    if (count > 0)
    {
        OnCountPropertyChanged();
        OnIndexerPropertyChanged();
        OnCollectionChanged(NotifyCollectionChangedAction.Remove, removedItems, index);
    }
}
```

Reply:
I thought about this some more and I think I am going to implement the 2nd option. This will allow any additional validation to in the base class of `Collection<T>`

```c#
protected override void RemoveItemsRange(int index, int count)
{
    // omitted code
  
    if (count > 0)
    {
        OnCountPropertyChanged();
        OnIndexerPropertyChanged();
        OnCollectionChanged(NotifyCollectionChangedAction.Remove, removedItems, index);
    }
}
```